src/core/net/ip6.cpp

```diff
@@ -627,6 +629,23 @@ ThreadError Ip6::ProcessReceiveCallback(const Message &aMessage, const MessageIn
     mReceiveIp6DatagramCallback(messageCopy, mReceiveIp6DatagramCallbackContext);
 
 exit:
+
+    switch (error)
+    {
+    case kThreadError_NoBufs:
```

Comment:
Is this supposed to test the `NoRoute` error condition? I don't see how `NoBufs` is possible.